localstack/services/awslambda/event_source_listeners/stream_event_source_listener.py

```diff
@@ -198,63 +200,75 @@ def _listen_to_shard_and_invoke_lambda(self, params: Dict):
-                                    shard_id,
-                                    first_seq_num,
-                                    last_seq_num,
-                                    stream_arn,
-                                    function_arn,
-                                    num_invocation_failures,
-                                    status_code,
-                                    batch_size,
-                                    first_arrival_time,
-                                    last_arrival_time,
-                                    failure_destination,
-                                )
                         else:
-                            should_get_next_batch = False
-                if should_get_next_batch:
-                    shard_iterator = records_response["NextShardIterator"]
-                    num_invocation_failures = 0
+                            num_invocation_failures += 1
+                            if num_invocation_failures >= max_num_retries:
+                                should_get_next_batch = True
+                                if failure_destination:
+                                    first_rec = records[0]
+                                    last_rec = records[-1]
+                                    (
+                                        first_seq_num,
+                                        last_seq_num,
+                                    ) = self._get_starting_and_ending_sequence_numbers(
+                                        first_rec, last_rec
+                                    )
+                                    (
+                                        first_arrival_time,
+                                        last_arrival_time,
+                                    ) = self._get_first_and_last_arrival_time(first_rec, last_rec)
+                                    self._send_to_failure_destination(
+                                        shard_id,
+                                        first_seq_num,
+                                        last_seq_num,
+                                        stream_arn,
+                                        function_arn,
+                                        num_invocation_failures,
+                                        status_code,
+                                        batch_size,
+                                        first_arrival_time,
+                                        last_arrival_time,
+                                        failure_destination,
+                                    )
+                            else:
+                                should_get_next_batch = False
+                    if should_get_next_batch:
+                        shard_iterator = records_response["NextShardIterator"]
+                        num_invocation_failures = 0
                 time.sleep(self._POLL_INTERVAL_SEC)
         except Exception as e:
+            #
```

Comment:
comment missing or typo ?

Reply:
Whoopsie, that's not supposed to be in there. Will remove!